configure.ps1

```diff
@@ -129,20 +129,23 @@ if (-not $ProgramFiles -or -not (Test-Path $ProgramFiles)) {
 $MSBuildDefaultRoot = Get-MSBuildRoot
 $MSBuildRelativePath = 'bin\msbuild.exe'
 
-Invoke-BuildStep 'Validating VS15 toolset installation' {
-    $vs15 = New-BuildToolset 15
+Invoke-BuildStep 'Validating VS toolset installation' {
+
+    $vsMajorVersion = Get-VSMajorVersion
```

Comment:
I think there's another case that can be handled. 

Say you only have VS16 on your machine, and you are not running this in developer command prompt. 
 
You'd need to pass special switches to force configure.ps1 to pick it up. 

Ideally when discovering the VS installation, you can return a tuple of (version, msbuildpath). 
Then based on which version was discovered here, validate the toolset.

Reply:
>Say you only have VS16 on your machine, and you are not running this in developer command prompt.

This will work in that case as well because while building toolset object, we also use vswhere to find the latest VS installation and use that. Corresponding code at https://github.com/NuGet/NuGet.Client/blob/dev/configure.ps1#L94